### PR TITLE
Trace Insurance changes from BackOffice site

### PR DESCRIPTION
**Pull request by dkaplunov-fd**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java [hunk 1]
```diff
@@ -263,7 +265,7 @@ public void on(InsuranceMandateCreatedEvent mandateCreatedEvent) {
   }
 
   @EventHandler
-  public void on(ActivationDateUpdatedEvent e) {
+  public void on(ActivationDateUpdatedEvent e, @MetaDataValue("token") String token) {
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java]: are we using the token here? if no, shall we remove ii?

Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java]: Yes, we receive token here from "apply" method in aggregate and save it in MetaData. There is three points to tracing: activation Date, cancellation Date and Certificate upload.

File: src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java [hunk 2]
```diff
@@ -272,17 +274,19 @@ public void on(ActivationDateUpdatedEvent e) {
...
 
     final ProductEntity pe = repository.findById(e.getProductId()).orElse(null);
+
     pe.activeFrom = e.getActivationDate().atZone(ZoneId.of("Europe/Stockholm")).toLocalDateTime();
     repository.save(pe);
   }
 
   @EventHandler
-  public void on(CancellationDateSetEvent e) {
+  public void on(CancellationDateSetEvent e, @MetaDataValue("token") String token) {
```
[the remark this answers is not in the document]
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java]: same

File: src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java [hunk 3]
```diff
@@ -312,7 +316,7 @@ public void on(MemberTerminatedEvent e) {
   }
 
   @EventHandler
-  public void on(CertificateUploadedEvent e) {
+  public void on(CertificateUploadedEvent e, @MetaDataValue("token") String token) {
```
[the remark this answers is not in the document]
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/productPricing/service/query/ProductEventListener.java]: same

File: src/main/java/com/hedvig/productPricing/service/service/TraceProductHelper.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  private static String formatDate (Date date) {
+    SimpleDateFormat sdf = new SimpleDateFormat("dd MMMM YYYY");
+    return sdf.format(date);
+  }
+
+  private static String formatDate (Instant date) {
+    return formatDate(Date.from(date));
+  }
+
+  public static String getUserId (String token) {
```
[the remark this answers is not in the document]
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/productPricing/service/service/TraceProductHelper.java]: done

**Pull request merged**